.github/workflows/charm-update-libs.yaml

```diff
@@ -0,0 +1,95 @@
+name: Update Charm Libraries
+on:
+  schedule:
+    # Weekly on Monday at 09:00 UTC
+    - cron: "0 9 * * 1"
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+env:
+  GIT_BRANCH: "auto/charm-libs"
+  CHARMCRAFT_CHANNEL: "latest/stable"
+
+jobs:
+  update-lib:
+    name: Check charm libraries
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash --noprofile --norc -euo pipefail {0}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Install charmcraft
+        run: |
+          sudo snap install charmcraft --classic --channel "${{ env.CHARMCRAFT_CHANNEL }}"
+
+      - name: Fetch charm libraries
+        id: fetch-libs
+        run: |
+          charmcraft fetch-lib
+
+      - name: Check diff
+        id: check-diff
+        run: |
+          changed_files=$(git diff --name-only && git ls-files --others --exclude-standard)
+
+          # Initialize variables
+          invalid_files=""
+          updated_libs=""
+
+          # Ensure only files in lib/charms/ are modified
+          for file in $changed_files; do
+            if [[ "$file" == "lib/charms/"* ]]; then
+              # File matches the pattern for charm libraries
+              updated_libs+="$file"$'\n'
+            else
+              # File does not match the pattern
+              invalid_files+="$file"$'\n'
+              echo "::error file=$(realpath "$file"),title=Unexpected-Diff::The file '$file' was modified, but it is not a charm library file."
+            fi
+          done
+
+          # Abort if invalid files were changed
+          if [[ -n "$invalid_files" ]]; then
+            exit 1
+          fi
+
+          # Set output variables
+          echo "updated_libs<<EOF" >> $GITHUB_OUTPUT
+          echo "$updated_libs" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
```

Comment:
That particular comment is unnecessary but you might want to explain the reason for the EOF trick.